warp/cairo-src/evm/bit_packing.cairo

```diff
@@ -1,13 +1,19 @@
+from starkware.cairo.common.bitwise import bitwise_and, bitwise_not
+from starkware.cairo.common.cairo_builtins import BitwiseBuiltin
 from starkware.cairo.common.math import assert_lt, unsigned_div_rem
 from starkware.cairo.common.pow import pow
-from starkware.cairo.common.uint256 import Uint256, uint256_shl
 
-func extract_unaligned_uint128{range_check_ptr}(shift, low, high) -> (value):
+const UINT128_BOUND = 2 ** 128
+
+func extract_unaligned_uint128{range_check_ptr, bitwise_ptr : BitwiseBuiltin*}(
+        shift, low, high) -> (value):
     # Given two aligned uint128's, extract an unaligned uint128,
     # shifted by 'shift' bytes from the 'high' high end.
-    let x = Uint256(low=low, high=high)
-    let (y) = uint256_shl(x, Uint256(low=8 * shift, high=0))
-    return (y.high)
+    let (p) = pow(256, 16 - shift)
+    let (low_mask) = bitwise_not(p - 1)
```

Comment:
will extracting `p - 1` here going to have any performance implication, I don't imagine that the cairo-compiler can do that kind of optimisation.

Reply:
Not much effect, probably, just one more memory cell, one less operation. Even one less felt division in this function gave me about just ~15 steps less.

Reply:
Okay